js/tabRestore.js

```diff
@@ -0,0 +1,33 @@
+const MAX_TAB_HISTORY_DEPTH = 20
+var tabHistoryStack = Array()
+
+var tabHistory = {
+  push: function (tabId) {
+    var tab = tabs.get(tabId)
+    // Do not store private tabs or blank tabs
+    if (tab.private || tab.url === 'about:blank' || tab.url === '') {
+      return
+    }
+
+    if (tabHistoryStack.length < MAX_TAB_HISTORY_DEPTH) {
+      tabHistoryStack.push(tab)
+    } else {
+      tabHistoryStack.shift()
+      tabHistoryStack.push(tab)
+    }
+  },
+  restore: function () {
+    if (tabHistoryStack.length === 0) {
+      return
+    }
+
+    var newIndex = tabs.getIndex(tabs.getSelected()) + 1
+    var newTab = tabs.add(tabHistoryStack.pop(), newIndex)
+
+    addTab(newTab, {
```

Comment:
Good catch.

> For example, if the tab was closed in task A but you're in task B now you should switch to task A before restoring.
Agreed 100%

> And if task A has been closed meanwhile you'd ask the user wheter they want the tab in task B or reopen task A with that tab.

The problem I see with this is that electron offers no prompt dialog in its API. It might be more trouble than it's worth to implement that nicely on all platforms. I think we should just assume that the user meant to restore the task they destroyed.